mlflow/telemetry/track.py

```diff
@@ -68,11 +72,15 @@ def _generate_telemetry_record(
         if params and params[0] == "cls" and isinstance(arguments["cls"], type):
             del arguments["cls"]
 
-        record_params = event.parse(arguments)
+        record_params = event.parse(arguments) or {}
+        if event == CreateExperimentEvent:
+            record_params.update(event.parse_result(result))
```

Comment:
nit: i wonder if we should make it more general in case we want to extend it in the future? like
```suggestion
        if hasattr(event, "parse_result"):
            record_params.update(event.parse_result(result))
```

it's also fine if we want to play it safe and not generalize for now, just curious